src/main/scala/com/gu/newsletterlistcleanse/models/NewsletterCutOff.scala

```diff
@@ -5,13 +5,25 @@ import java.time.ZonedDateTime
 import io.circe.{Encoder, Decoder}
 import io.circe.generic.semiauto.{deriveEncoder, deriveDecoder}
 
+case class BrazeData (
+  brazeSubscribeAttributeName: String,
+  brazeSubscribeEventNamePrefix: String
+)
+
+object BrazeData {
+  implicit val newsletterCutOffEncoder: Encoder[BrazeData] = deriveEncoder
+  implicit val newsletterCutOffDecoder: Decoder[BrazeData] = deriveDecoder
+}
 case class NewsletterCutOff(
   newsletterName: String,
   cutOffDate: ZonedDateTime,
-  activeListLength: Int
+  activeListLength: Int,
+  brazeData: BrazeData
 )
 
 object NewsletterCutOff {
+  def apply(newsletterName: String, cutOffDate: ZonedDateTime, activeListLength: Int) : (BrazeData => NewsletterCutOff)
```

Comment:
Ha, this threw me off later when I saw `NewsetterCutOff(args...)` being called because I was expecting it to be an object construction. Here `apply` isn't used as a constructor that returns the type `NewsletterCutOff`. Would you mind giving this method a name such that it can be called using `NewsletterCutOff.forBrazeData` (or whatever name is appropriate)

I think this comment is more a reflection of my personal taste as your code is perfectly valid, and I'm sure asking around some people might disagree with me.

Reply:
I had the same problem and couldn't get it to work how you suggested, but I think that may have been to me conflating this change with another pre-existing bug. I'll look at implementing it as you suggest as part of a general refactor off the back of this